repos/spack_repo/builtin/packages/visit/package.py

```diff
@@ -196,6 +196,7 @@ class Visit(CMakePackage):
     depends_on("mfem@4.4:", when="+mfem")
     depends_on("mfem+shared+exceptions+fms+conduit", when="+mfem")
     depends_on("libfms@0.2:", when="+mfem")
+    conflicts("^mfem build_system=cmake")
```

Comment:
I see this `conflicts()` line in `visit/package.py` and a few others. I guess it is telling Spack, if someone builds mfem with CMake, the resulting MFEM install cannot be used within these other packages? If so, I am wondering what difficulty is arising as a result? I mean, is MFEM's CMake install creating cmake content that is causing other packages to digest it incorrectly? If the rational could be elaborated a tad here, that might be useful.

Reply:
I think @jacobmerson can clarify, but from what I can tell, MFEM has two builds that are supported: the CMake build, and the makefile based build. Both builds export different build objects, and do depending on how the parent project handles those builds, they make / break accordingly.

Reply:
Ah, ok. That makes sense. So, in the case of VisIt, the `package.py` appears to have logic that is dependent upon MFEM having been built and installed using the vanilla GNU make approach.

Reply:
So, maybe this is NOT the time/place to inquire about this but in looking at VisIt's `package.py` file, I honestly can't see anything there that would suggest a dependency in the build approach used for MFEM. I searched for all instances of `mfem`, and all I see are `variant()`, `depends_on()` statements and some logic that turns around and a logic block conditional on `+mfem` being the `spec`, that defines variables relevant to VisIt's build.

So, I honestly can't see any reason why VisIt's `package.py` file is somehow dependent on MFEM's build/installation. Do I have the wrong expectations of Spack here?

Reply:
@cameronrutherford is correct the packages that I added `conflict` statements to utilize build artifacts that are generated by the `Makefiles`. This constraint that the `Makefile` based artifacts are implicit in the `Makefiles` of the downstream packages like `visit`. The addition of the `conflicts` statement makes this constraint explicit for the `spack` package. Previously, `spack` could only build using the `Makefile` based approach so it was not necessary to have any explicit specification of the build system in the package.

Reply:
I'm not convinced this conflict is actually needed for the `visit` package. @jacobmerson, did you actually try building `visit` with CMake built MFEM?

Reply:
I finally had a chance to test this out in a clean VM. looks like visit works with the `CMake` build. I'll remove the conflict.